glotaran/parameter/parameter.py

```diff
@@ -343,152 +439,181 @@ def __setstate__(self, state):
-        return self._getval() == other
+        return self._value == other
 
-    def __ne__(self, other):
+    def __ne__(self, other):  # noqa D400
         """!="""
-        return self._getval() != other
+        return self._value != other
 
-    def __radd__(self, other):
+    def __radd__(self, other):  # noqa D400
         """+ (right)"""
-        return other + self._getval()
+        return other + self._value
 
-    def __rtruediv__(self, other):
+    def __rtruediv__(self, other):  # noqa D400
         """/ (right)"""
-        return other / self._getval()
+        return other / self._value
 
-    def __rdivmod__(self, other):
+    def __rdivmod__(self, other):  # noqa D400
         """divmod (right)"""
-        return divmod(other, self._getval())
+        return divmod(other, self._value)
 
-    def __rfloordiv__(self, other):
+    def __rfloordiv__(self, other):  # noqa D400
         """// (right)"""
-        return other // self._getval()
+        return other // self._value
 
-    def __rmod__(self, other):
+    def __rmod__(self, other):  # noqa D400
         """% (right)"""
-        return other % self._getval()
+        return other % self._value
 
-    def __rmul__(self, other):
+    def __rmul__(self, other):  # noqa D400
         """* (right)"""
-        return other * self._getval()
+        return other * self._value
 
-    def __rpow__(self, other):
+    def __rpow__(self, other):  # noqa D400
         """** (right)"""
-        return other ** self._getval()
+        return other ** self._value
 
-    def __rsub__(self, other):
+    def __rsub__(self, other):  # noqa D400
         """- (right)"""
-        return other - self._getval()
+        return other - self._value
 
 
-def _log_value(value: float):
+def _log_value(value: float) -> float:
+    """Get the logarithm of a value.
+
+    Performs a check for edge cases and migitates numerical issues.
+
+    Parameters
+    ----------
+    value : float
+        The initial value.
+
+    Returns
+    -------
+    float
+        The logarithm of the value.
+    """
     if not np.isfinite(value):
         return value
     if value == 1:
         value += 1e-10
     return np.log(value)
 
 
-def _retrieve_from_list_by_type(li: list, t: type | tuple[type, ...], default: Any):
-    tmp = list(filter(lambda x: isinstance(x, t), li))
+def _retrieve_item_from_list_by_type(
+    item_list: list, item_type: type | tuple[type, ...], default: Any
+) -> Any:
+    """Retrieve an item from list which matches a given type.
+
+    Parameters
+    ----------
+    item_list : list
+        The list to retrieve from.
+    item_type : type | tuple[type, ...]
+        The item type or tuple of types to match.
+    default : Any
+        Returned if no item matches.
+
+    Returns
+    -------
+    Any
+
+    """
+    tmp = list(filter(lambda x: isinstance(x, item_type), item_list))
     if not tmp:
         return default
-    li.remove(tmp[0])
+    item_list.remove(tmp[0])
```

Comment:
What's going on here? Based on the function signature and documentation I would not expect this function to have this behavior.